### DISCUSSION TITLE
Is it possible to run transformations to JSX / TS in the browser using swc via wasm?

### DISCUSSION THREAD

**Discussion by YarivGilad** [swc-project/swc#1520] [Q&A] [answered]
Hi all, 
I'm after creating an **online code editor** using Monaco / Ace as the editor component, and in order to support compilation/transformation of JSX or TS, I was wondering if there is a way to use SWC in the browser using web assembly rather than the node.js process. 
I'd very much appreciate your inputs. 
Also, if you know of an existing editor that achieved this goal, I'll be happy to know about it.
Thank you!
Yariv

**Comment by kdy1** [comment, marked as answer]
It's possible.
Just use `@swc/wasm`

**Reply by YarivGilad** [reply]
Sweet :)
Just wanted to get some sort of affirmation before I'm embarking on this endeavor... 
Thank you!

**Reply by YarivGilad** [reply]
Is there any documentation on @swc/wasm?

**Reply by kdy1** [reply]
You can use `transformSync` and `transform` with same api.
If it doesn't work identically, please file an issue. (Except ones related to paths)

**Reply by YarivGilad** [reply]
will do. thanks again!

**Reply by YarivGilad** [reply]
Just making sure I'm doing it as intended...
I was able to require @swc/core and @swc/wasm in node.js and use the swc.transformSync() on both, while swc.transform() worked only when requiring @swc/core and not @swc/wasm. 
More critical for me is a first grasp on how to run the transformation within the browser rather then node.js...
Am I suppose to use a script tag with type module and src of 'https://unpkg.com/@swc/wasm@1.2.51/wasm.js' ?
I'm a newb with WA...
